apps/heft/src/cli/actions/HeftActionBase.ts

```diff
@@ -17,20 +17,24 @@ import { PluginManager } from '../../pluginFramework/PluginManager';
 /**
  * @public
  */
-export interface IActionDataBase<THooks extends ActionHooksBase> {
+export interface IActionContext<
+  THooks extends ActionHooksBase<TActionProperties>,
+  TActionProperties extends object
+> {
   hooks: THooks;
+  properties: TActionProperties;
 }
 
 /**
  * @public
  */
-export abstract class ActionHooksBase {
+export abstract class ActionHooksBase<TActionProperties extends object> {
   /**
    * This hook allows the action's execution to be completely overridden. Only the last-registered plugin
```

Comment:
The semantics here seem potentially muddy. Perhaps it should be an error if more than one plugin tries to do this?  Perhaps this is not an event handler at all, but some other kind of contract?

Reply:
Let's mark this as `@beta` for now and tackle this later as the API matures.